ext/cairo/rb_cairo_font_face.c

```diff
@@ -108,6 +117,11 @@ rb_cairo_font_face_to_ruby_object (cairo_font_face_t *face)
 static VALUE
 cr_font_face_allocate (VALUE klass)
 {
+#if CAIRO_HAS_FT_FONT
+  FT_Error error = FT_Init_FreeType(&library);
```

Comment:
> Why would the allocate function be called multiple times per instance?

The allocate function is called just once for one (Ruby) instance. I said that `FT_init_FreeType(&library)` is called multiple times because Ruby user can create multiple font Ruby instances. The same `library` (C) object is initialized multiple times. Is it OK?

> Shouldn't that static instance be different per object instance the same way that `ruby_object_key` is?

It's OK when the static instance can be shared by multiple objects. `ruby_object_key` can be shared by multiple object. Because cairo just uses the address of `ruby_object_key`. The address isn't changed.

Again, I said that calling `FT_Init_FreeType(&library)` multiple times is a problem.